presto-main/src/main/java/com/facebook/presto/Session.java

```diff
@@ -294,6 +298,16 @@ public AccessControlContext getAccessControlContext()
         return context;
     }
 
+    public void setOptimizedWithMaterializedView()
```

Comment:
We might want to add many other counters later. Instead of add all the set functions to Session, it might be better to move the logic to the caller, i.e., the caller use getStats and add the counter.